File: park-admin-app/park-admin-webapp/src/main/web/src/app/model/nivel.ts
import {Conjunto} from './conjunto';


export class Nivel {
  numero: number;
  orientacion: Orientacion;
  conjuntos: Conjunto[];

  constructor(numero: number) {
    this.numero = numero;
    this.orientacion = Orientacion.Horizontal;
    this.conjuntos = [];
  }

}


enum Orientacion {
  Horizontal,
  Vertical,
}

File: park-admin-app/park-admin-server/src/main/java/com/parkadmin/parkadminserver/model/Nivel.java
package com.parkadmin.parkadminserver.model;

import lombok.Data;

import javax.persistence.*;
import java.util.List;

@Data
@Entity
public class Nivel {
    @Id
    private Long numero;
    private Orientacion orientacion;
    @ManyToOne
    private Estacionamiento estacionamiento;
    @OneToMany(cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Espacio> espacios;
}

enum Orientacion {
    HORIZONTAL,
    VERTICAL
}